File: sdks/ruby/lib/backlex.rb
# frozen_string_literal: true

# Official Ruby client for the backlex API.
#
#   require "backlex"
#   client = Backlex::Client.new("https://api.example.com", api_key: "pak_...")
#   posts = client.from("posts").query.where(Backlex::Filter.eq("published", true)).list
module Backlex
  VERSION = "0.0.1"
end

require_relative "backlex/error"
require_relative "backlex/filter"
require_relative "backlex/query_builder"
require_relative "backlex/collection"
require_relative "backlex/auth"
require_relative "backlex/storage"
require_relative "backlex/realtime"
require_relative "backlex/client"

File: sdks/dart/lib/backlex.dart
/// Official Dart client for the backlex API.
///
///     import 'package:backlex/backlex.dart';
///
///     final client = Client('https://api.example.com', apiKey: 'pak_...');
///     final posts = await client.from('posts').query()
///         .where(Filter.eq('published', true)).list();
library backlex;

import 'dart:async';
import 'dart:convert';
import 'dart:io';
import 'dart:math';

part 'src/error.dart';
part 'src/filter.dart';
part 'src/query_builder.dart';
part 'src/collection.dart';
part 'src/auth.dart';
part 'src/storage.dart';
part 'src/realtime.dart';
part 'src/client.dart';
